content/en/flux/cheatsheets/sharding.md

```diff
@@ -0,0 +1,325 @@
+---
+title: "Controller sharding and horizontal scaling"
+linkTitle: "Sharding"
+description: "Showcase how to configure sharding of Flux controllers at bootstrap time."
+weight: 30
+---
+
+When Flux is managing tens of thousands of applications, it is advised to adopt
+a sharding strategy to spread the load between multiple instances of Flux controllers.
+To enable horizontal scaling, each controller can be deployed multiple times
+with a unique label selector which is used as the sharding key.
+
+What follows is a guide on how to bootstrap multiple controller instances and how to
+shard the reconciliation of Flux resources using the `sharding.fluxcd.io/key` label.
+
+## Bootstrap with sharding
+
+At [bootstrap time](bootstrap.md), you can define the number of shards
+and spin up a Flux controller instance for each shard.
+
+First you'll need to create a Git repository and clone it locally, then
+create the file structure required by bootstrap with:
+
+```sh
+mkdir -p clusters/my-cluster/flux-system
+touch clusters/my-cluster/flux-system/gotk-components.yaml \
+    clusters/my-cluster/flux-system/gotk-sync.yaml \
+    clusters/my-cluster/flux-system/kustomization.yaml
+```
+
+Then you'll create a dedicated directory inside `flux-system` for each shard:
+
+```sh
+mkdir -p clusters/my-cluster/flux-system/shard1
+touch clusters/my-cluster/flux-system/shard1/kustomization.yaml
+```
+
+### Configure controller sharding
```

Comment:
I feel all these steps should rather be a numbered list. Alternatively enumerate the headings like `### 2. Configure controller sharding`